extension/src/popup/components/signTransaction/Operations.tsx

```diff
@@ -368,40 +373,28 @@ export const Operations = ({
               ) : null}
 
               {signer?.ed25519PublicKey ? (
-                <>
-                  <KeyValueWithPublicKey
-                    operationKey="Signer"
-                    operationValue={signer.ed25519PublicKey}
-                  />
-                  <KeyValueList
-                    operationKey="Weight"
-                    operationValue={signer.weight}
-                  />
-                </>
+                <KeyValueWithPublicKey
+                  operationKey="Signer"
+                  operationValue={signer.ed25519PublicKey}
+                />
               ) : null}
               {signer?.sha256Hash ? (
-                <>
-                  <KeyValueList
-                    operationKey="Signer"
-                    operationValue={formattedBuffer(signer?.sha256Hash?.data)}
-                  />
-                  <KeyValueList
-                    operationKey="Weight"
-                    operationValue={signer.weight}
-                  />
-                </>
+                <KeyValueList
+                  operationKey="Signer"
+                  operationValue={formattedBuffer(signer?.sha256Hash?.data)}
+                />
               ) : null}
               {signer?.preAuthTx ? (
-                <>
-                  <KeyValueList
-                    operationKey="Signer"
-                    operationValue={formattedBuffer(signer.preAuthTx.data)}
-                  />
-                  <KeyValueList
-                    operationKey="Weight"
-                    operationValue={signer.weight}
-                  />
-                </>
+                <KeyValueList
+                  operationKey="Signer"
+                  operationValue={formattedBuffer(signer.preAuthTx.data)}
+                />
+              ) : null}
+              {signer?.weight ? (
+                <KeyValueList
+                  operationKey="Weight"
+                  operationValue={signer.weight}
+                />
```

Comment:
check for signer.weight on its own rather then trying to predict when me may need to show it